### DISCUSSION TITLE
Load Chart dynamicly

### DISCUSSION THREAD

**Discussion by akvaskov** [icehouse-ventures/laravel-chartjs#15] [General]
Hello! I have a page which loads and shows content dynamicly. Say I have a button which loads a ChartJS. Unfortunatelly, as js script, which builds a chart, is loaded inside canvas tag, it doesn't execute after loading althouth I see in the console that the chart was rendered. Any ideas how to make js script run?

**Comment by peterjthomson** [comment]
If you have JavaScript already running on the page then this package isn't really much help as you'll already have a JavaScript build pipeline and/or the script tags. If you find the php friendly array format in the package helpful but want to go for more advanced JavaScript (or Livewire) patterns then you can publish the blade view which in turn exposes the on-page Javascript tag as editable JS that you can directly modify.

To deal with your particular situation it sounds like you might want an async await or a listener on the Chart and/or on your custom JS. You could source dive the Livewire view file to get some inspiration on how we handled wiring the Chart up with Alpine JS (the JS part of Livewire).
